src/components/step-flow/step-flow.component.tsx

```diff
@@ -0,0 +1,193 @@
+import React from "react";
+import { MarginProps } from "styled-system";
+import Icon from "../icon";
+import IconButton from "../icon-button";
+import {
+  StyledStepFlow,
+  StyledStepContent,
+  StyledStepContentText,
+  StyledCategoryText,
+  StyledTitleText,
+  StyledTitleTextContainer,
+  StyledVisibleTitleText,
+  StyledHiddenTitleText,
+  StyledStepLabelAndProgress,
+  StyledStepLabel,
+  StyledProgressIndicatorBar,
+  StyledProgressIndicator,
+} from "./step-flow.style";
+import tagComponent, {
+  TagProps,
+} from "../../__internal__/utils/helpers/tags/tags";
+import { VariantTypes } from "../typography";
+import useLocale from "../../hooks/__internal__/useLocale";
+
+export type Steps = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8;
+
+export interface StepFlowProps extends MarginProps, TagProps {
+  /** A category for the user journey.  */
+  category?: string;
+  /** Set the variant of the internal 'Typography' component which contains the category.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  categoryTypographyVariant?: VariantTypes;
+  /** The title of the current step.  */
+  title: string;
+  /** Set the variant of the internal 'Typography' component which contains the title.
+   * However, despite the chosen variant the styling will always be overridden.
+   */
+  titleTypographyVariant?: VariantTypes;
```

Comment:
```suggestion
  titleVariant? :Extract<VariantTypes, "h1" | "h2" | "h3" | "h4" | "h5" | "p">;
```

Naming is non-blocking for me but I think we should limit the elements here as some don't seem viable for this use case